### PR TITLE
Search Controller & Form to search by Location

### PR DESCRIPTION
**Pull request by MarvinCrepin**

### CODE REVIEW

**Comment by MarjorieThomassin** [comment]
You should add method get on your annotation

Requests about spaces must go on SpaceController and not SearchController

**Comment by MarvinCrepin** [comment]
> You should add method get on your annotation
> 
> Requests about spaces must go on SpaceController and not SearchController

done

File: src/Controller/SearchController.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            return $this->redirectToRoute('app_space_list', [
+                'location' => $location,
+            ]);
+        }
+
+        return $this->renderForm('search/index.html.twig', [
+            'form' => $form,
+        ]);
+    }
+}
```
Comment (rocambille) [inline src/Controller/SearchController.php]: this controller is not needed: this logic should be in SpaceController

File: src/Controller/SpaceController.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+class SpaceController extends AbstractController
+{
+
+    /**
+     * @Route("/space/{location}", name="app_space_list")
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: location should be part of the query string, not an url parameter

**Review by rocambille — CHANGES_REQUESTED** [review]

File: src/Controller/SpaceController.php (hunk 2 above)
Comment (MarvinCrepin) [reply src/Controller/SpaceController.php]: done

File: src/Controller/SpaceController.php [hunk 3]
```diff
@@ -13,11 +13,34 @@ class SpaceController extends AbstractController
...
+
+            return $this->redirectToRoute('app_space_list', [
+                'location' => $location,
+            ]);
+        }
+
+        return $this->renderForm('search/index.html.twig', [
+            'form' => $form,
+        ]);
+    }
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: this should be mixed with index method (rebase your branch on a pulled master)

File: src/Controller/SpaceController.php [hunk 4]
```diff
@@ -13,11 +13,34 @@ class SpaceController extends AbstractController
...
+
+        $form = $this->createForm(SearchType::class, null, array('method' => 'GET'));
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $location = $form->get('location')->getData();
+
+            return $this->redirectToRoute('app_space_list', [
+                'location' => $location,
+            ]);
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: once mixed with index, no need to redirect: change the value of the rendered space list

**Review by rocambille — CHANGES_REQUESTED** [review]

File: src/Controller/SpaceController.php (hunk 4 above)
Comment (MarvinCrepin) [reply src/Controller/SpaceController.php]: done

File: src/Controller/SpaceController.php [hunk 5]
```diff
@@ -42,6 +56,19 @@ public function new(Request $request, EntityManagerInterface $entityManager): Re
...
+
+    public function list(SpaceRepository $spaceRepository, Request $request): Response
+    {
+        $location = $request->get('location');
+
+        $space = $spaceRepository->findByLocation($location);
+        return $this->render('search/list.html.twig', [
+            'space' => $space,
+        ]);
+    }
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: you can remove this one

File: templates/space/list.html.twig [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{% for space in spaces %}
+
+<p>{{ space.name }}</p>
+{% endfor %}
```
Comment (rocambille) [inline templates/space/list.html.twig]: move this in index, and remove this file

File: templates/search/index.html.twig [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+    <div class="row">
+        <div class="col">
+            {{ form_row(form.location) }}
+        </div>
+        <div class="col" id="some-custom-id">
+        </div>
+    </div>
+{{ form_end(form) }}
+{% endblock %}
```
Comment (rocambille) [inline templates/search/index.html.twig]: remove the whole directory templates/search

File: src/Controller/SpaceController.php [hunk 8]
```diff
@@ -11,15 +12,28 @@
...
 
-#[Route('/space')]
 class SpaceController extends AbstractController
 {
-    #[Route('/', name: 'space_index', methods: ['GET'])]
-    public function index(SpaceRepository $spaceRepository): Response
+
+    /**
+     * @Route("/", name="app_home")
+     */
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: keep the previous route declaration:

    #[Route('/', name: 'space_index', methods: ['GET'])]

File: src/Controller/SpaceController.php [hunk 9]
```diff
@@ -11,15 +12,28 @@
...
+        $form = $this->createForm(SearchType::class, null, array('method' => 'GET'));
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $location = $form->get('location')->getData();
+
+            return $this->render('space/list.html.twig', [
+                'location' => $location,
+                'spaces' => $spaceRepository->findByLocation($location)
+            ]);
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: remove this return

File: src/Controller/SpaceController.php [hunk 10]
```diff
@@ -11,15 +12,28 @@
...
+     */
+    public function index(Request $request, SpaceRepository $spaceRepository): Response
     {
-        return $this->render('space/index.html.twig', [
-            'spaces' => $spaceRepository->findAll(),
-        ]);
+
+        $form = $this->createForm(SearchType::class, null, array('method' => 'GET'));
+        $form->handleRequest($request);
+
```
Comment (rocambille) [inline src/Controller/SpaceController.php:25]: add here line to have a default value for $spaces:

    $spaces = $spaceRepository->findAll();

File: src/Controller/SpaceController.php [hunk 11]
```diff
@@ -11,15 +12,28 @@
...
+        if ($form->isSubmitted() && $form->isValid()) {
+            $location = $form->get('location')->getData();
+
```
Comment (rocambille) [inline src/Controller/SpaceController.php:28]: add here line to update value for $spaces:

    $spaces = $spaceRepository->findByLocation($location);

File: src/Controller/SpaceController.php [hunk 12]
```diff
@@ -11,15 +12,28 @@
...
+            return $this->render('space/list.html.twig', [
+                'location' => $location,
+                'spaces' => $spaceRepository->findByLocation($location)
+            ]);
+        }
+
+        return $this->renderForm('space/index.html.twig', ['form' => $form]);
```
Comment (rocambille) [inline src/Controller/SpaceController.php]: add $spaces and $location to the array (and handle in the template that location may be null)

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by MarvinCrepin** [comment]
done

**Pull request merged**